synapse/util/templates.py

```diff
@@ -56,7 +56,8 @@ def build_jinja_env(
     if autoescape is None:
         autoescape = jinja2.select_autoescape()
 
-    loader = jinja2.FileSystemLoader(template_search_directories)
+    # the type signature of this is wrong
+    loader = jinja2.FileSystemLoader(template_search_directories)  # type: ignore[arg-type]
```

Comment:
Is it? It looks like it should accept a `Sequence[str]`? Which suggests *our* types our wrong

Reply:
True. I suppose the types are an interface detail, and the actual implementation-accepted types can change. Will address.

Reply:
I'm don't understand what you mean there, sorry?

Reply:
Jinja2 currently accepts `Iterable[str]` but says it will accept `Sequence[str]`.
So morally speaking, they might change the implementation that breaks something so that `Iterable[str]` doesn't work anymore, and it would be silly us for not trusting their type annotations.
(Hence I've converted our iterator to a `List`.)